### ISSUE TITLE
[no-unused-vars] false positive for imported type alias

### ISSUE THREAD

**Issue by cdalexndr** [typescript-eslint/typescript-eslint#1678]
**Repro**

```JSON
env:
  browser: true
  es6: true
extends:
  - 'eslint:recommended'
  - 'plugin:@typescript-eslint/eslint-recommended'
globals:
  Atomics: readonly
  SharedArrayBuffer: readonly
parser: '@typescript-eslint/parser'
parserOptions:
  ecmaVersion: 2018
  sourceType: module
plugins:
  - '@typescript-eslint'
rules: {}

```

```TS
import DateTimeFormatOptions = Intl.DateTimeFormatOptions;
function formatDate(date: Date, options?: DateTimeFormatOptions): string {
 ....
}
```
Same issue when importing classes with `import {CLASS} from "TSFILE";`

**Expected Result**
No error.

**Actual Result**
` error  'DateTimeFormatOptions' is defined but never used  no-unused-vars`

**Additional Info**
Related: https://github.com/typescript-eslint/typescript-eslint/issues/363

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.22.0` |
| `@typescript-eslint/parser`        | `2.22.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.13.1` |

**Comment by bradzacher** [comment]
duplicate of #1542
